proxy/proxy.go

```diff
@@ -32,6 +32,11 @@ func Middlewares(middlewares ...Middleware) func(r *http.Request) {
 }
 
 func MiddlewareSetUpstream(upstream *url.URL) Middleware {
+	// Verify that the middleware upstream is valid now; rather than failing during processing
+	if upstream == nil {
+		panic("no upstream")
```

Comment:
@esnible I think this is what is failing in the tests now (https://app.circleci.com/pipelines/github/observatorium/api/515/workflows/541d2f45-b811-4c09-8548-1909ab818240/jobs/2530?invite=true#step-102-886 )